### PR TITLE
Fix CqlExpressionVisitor when RuntimePropertyInfo

### PR DESCRIPTION
**Pull request by landim**
Linq evaluation throws exception when CqlExpressionVisitor tries to cast member to FieldInfo.
Reported on CSHARP-577

### CODE REVIEW

File: src/Cassandra/Data/Linq/CqlExpressionVisitor.cs [hunk 1]
```diff
@@ -1118,19 +1122,32 @@ private static object GetClosureValue(MemberExpression node)
...
+                case MemberTypes.Field:
+                    var fieldInfo = (FieldInfo)node.Member;
+                    getValueDel = fieldInfo.GetValue;
+                    break;
+                case MemberTypes.Property:
+                    var propertyInfo = (PropertyInfo) node.Member;
+                    getValueDel = propertyInfo.GetValue;
+                    break;
+                default:
+                    throw new NotSupportedException(node.Member.MemberType.ToString());
```
Comment (jorgebay) [inline src/Cassandra/Data/Linq/CqlExpressionVisitor.cs]: Use `CqlLinqNotSupportedException` instead.

File: src/Cassandra/Data/Linq/CqlExpressionVisitor.cs [hunk 2]
```diff
@@ -1118,19 +1122,32 @@ private static object GetClosureValue(MemberExpression node)
 
         private static object GetFieldValue(MemberExpression node)
         {
-            var fieldInfo = (FieldInfo)node.Member;
+            MemberOrPropertyGetValue getValueDel;
```
Comment (jorgebay) [inline src/Cassandra/Data/Linq/CqlExpressionVisitor.cs]: If you want to avoid creating the delegate, you can reuse the generic one: `Func<object, object>`

**Comment by jorgebay** [comment]
Besides the minor nits, the fix and test looks good!
It would be nice to could include tests for expressions with static properties, nested constants, ...

**Comment by landim** [comment]
There is an error while instantiating an object at where clause. I am looking into it.

**Comment by jorgebay** [comment]
Fixed in CSHARP-577.

**Pull request closed without merging**